plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -184,21 +184,30 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 	}
 
 	vid = fu_plugin_thunderbolt_udev_get_id (device, "vendor", &error);
-	if (vid == 0x0) {
+	if (vid == 0x0)
 		g_warning ("failed to get Vendor ID: %s", error->message);
-		return;
-	}
+
 	did = fu_plugin_thunderbolt_udev_get_id (device, "device", &error);
-	if (did == 0x0) {
+	if (did == 0x0)
 		g_warning ("failed to get Device ID: %s", error->message);
-		return;
-	}
 
 	is_host = fu_plugin_thunderbolt_is_host (device);
-	vendor_id = g_strdup_printf ("TBT:0x%04X", (guint) vid);
-	device_id = g_strdup_printf ("TBT-%04x%04x", (guint) vid, (guint) did);
+
+	version = g_udev_device_get_sysfs_attr (device, "nvm_version");
+	if (is_host && ((version == NULL) || (vid == 0x0) || (did == 0x0))) {
```

Comment:
I think we'll have to see what @ybernat comes back with from the lab setup on what all actually happens in safe mode.  Otherwise falling back to read is doable (just ugly).